### ISSUE TITLE
[v5] Okta SSO login not possible

### ISSUE THREAD

**Issue by PSchnurbus24** [reportportal/reportportal#1618]
**Describe the bug**
Login with Okta SSO not possible. HTTP Error 403.

**To Reproduce**
This probably happens when my Okta session is older than 24h.

**Expected behavior**
Successful login

**Screenshots**
Application log:
...
2021-12-21 09:56:21.361 DEBUG 1 --- [nio-9999-exec-2] o.s.s.w.u.matcher.AntPathRequestMatcher  : Checking match of request : '/saml/sp/SSO/alias/report-portal-sp'; against '/saml/sp/SSO/**'
2021-12-21 09:56:21.361 DEBUG 1 --- [nio-9999-exec-2] s.p.s.a.SamlAuthenticationResponseFilter : Request is to process authentication
2021-12-21 09:56:21.370 DEBUG 1 --- [nio-9999-exec-2] s.p.s.a.SamlAuthenticationResponseFilter : Authentication request failed: org.springframework.security.authentication.InsufficientAuthenticatio
nException: Validation Errors:
1. Authentication statement is too old to be used with value: '2021-12-20T07:15:27.544Z' current time: '2021-12-21T09:56:21.370Z'

org.springframework.security.authentication.InsufficientAuthenticationException: Validation Errors:
1. Authentication statement is too old to be used with value: '2021-12-20T07:15:27.544Z' current time: '2021-12-21T09:56:21.370Z'
        at org.springframework.security.saml.provider.service.authentication.SamlAuthenticationResponseFilter.attemptAuthentication(SamlAuthenticationResponseFilter.java:89) ~[spring-security-saml2
-core-2.0.0.M31.jar!/:2.0.0.M31]
        at org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter.doFilter(AbstractAuthenticationProcessingFilter.java:212) ~[spring-security-web-5.2.4.RELEASE.jar!/
:5.2.4.RELEASE]
        at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334) ~[spring-security-web-5.2.4.RELEASE.jar!/:5.2.4.RELEASE]
        at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:113) ~[spring-web-5.2.4.RELEASE.jar!/:5.2.4.RELEASE]
        at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334) ~[spring-security-web-5.2.4.RELEASE.jar!/:5.2.4.RELEASE]
...
2021-12-21 09:56:21.371 DEBUG 1 --- [nio-9999-exec-2] s.p.s.a.SamlAuthenticationResponseFilter : Updated SecurityContextHolder to contain null Authentication
2021-12-21 09:56:21.371 DEBUG 1 --- [nio-9999-exec-2] s.p.s.a.SamlAuthenticationResponseFilter : Delegating to authentication failure handler com.epam.reportportal.auth.AuthFailureHandler@61d01788
2021-12-21 09:56:21.371 DEBUG 1 --- [nio-9999-exec-2] w.c.HttpSessionSecurityContextRepository : SecurityContext is empty or contents are anonymous - context will not be stored in HttpSession.
2021-12-21 09:56:21.371 DEBUG 1 --- [nio-9999-exec-2] s.s.w.c.SecurityContextPersistenceFilter : SecurityContextHolder now cleared, as request processing completed
2021-12-21 09:56:21.372 ERROR 1 --- [nio-9999-exec-2] o.a.c.c.C.[.[.[/].[dispatcherServlet]    : Servlet.service() for servlet [dispatcherServlet] in context with path [] threw exception

**Additional context**
Installation in GKE cluster via helm. Version 5.6.0

**Comment by PSchnurbus24** [comment]
When I log out of Okta and log back in, the login in the reportportal is also successful again.

**Comment by beamerkun** [comment]
I had similar problem - real error was a bit earlier in the log, it contained `Destination mismatch`, which pointed me to #1474

I've added `RP_AUTH_SAML_BASE_PATH=https://<full-URL-to-reportportal>/uat` to authorization container vars and SAML SSO login started working again

**Comment by evjlobanova** [comment]
@VolhaKarenko @Yumfriez Could you please check if the problem on rp side?

**Comment by gabelilly** [comment]
I am having the same issue as described. It appears that if the Okta session was created more than 24 hours ago, the result is a 403 when redirected back to ReportPortal. I've tried setting RP_AUTH_SAML_BASE_PATH (both in the UI and in compose override) and increasing the session time using RP_SESSION_LIVE, but both did not solve this issue. Any further help would be very appreciated.

**Comment by VolhaKarenko** [comment]
hi @gabelilly, could you please provide detailed steps to reproduce the issue? 
Login to Okta -> wait 24 hours -> login to ReportPortal with Okta, have I understood you correctly?

**Comment by gabelilly** [comment]
Hello @VolhaKarenko! The steps you outlined are correct. Anytime the Okta session is less than 24 hours, SSO will work successfully. If the Okta session is older than 24 hours, Okta redirects to ReportPortal and the response is a 403.

**Comment by VolhaKarenko** [comment]
@gabelilly thanks for the details! We are currently investigating the issue, I will notify here about any updates on this

**Comment by evgenytkachenko** [comment]
It seems to me, the property RP_SESSION_LIVE in the UAT container does not make any effect. It has a value of 86400 and increasing the value does not help.

**Comment by beamerkun** [comment]
Problem lies with default values inside Spring framework SAML implementation. We had to override the defaults to match authentication session length of our SAML provider by adding this snippet somewhere inside `SamlServiceProviderBeanConfiguration` in `service-authorization`:
```
       @Bean
       public SamlValidator samlValidator() {
               DefaultValidator validator = new DefaultValidator(samlImplementation());
               validator.setMaxAuthenticationAgeMillis(14 * 24 * 60 * 60 * 1000);
               return validator;
       }
```

**Comment by VeranikaRymzha** [comment]
Hello everyone,
@PSchnurbus24 , @jsokol805, @gabelilly, @CraigHead 
FYI
The issue was fixed by the team: Saml live session will be configurable via env variable RP_SAML_SESSION-LIVE= #time in minutes.
This improvement will be a part of the next release of ReportPortal. We plan it for April.

**Comment by Kozovskaya** [comment]
Hi @PSchnurbus24 @jsokol805 @gabelilly @CraigHead,
the fix is already in our latest release 23.1!
https://reportportal.io/releases
